brut.apktool/apktool-lib/src/main/java/brut/androlib/res/AndrolibResources.java

```diff
@@ -457,6 +572,58 @@ public void aaptPackage(File apkFile, File manifest, File resDir, File rawDir, F
         }
     }
 
+    public void aaptPackage(File apkFile, File manifest, File resDir, File rawDir, File assetDir, File[] include)
+            throws AndrolibException {
+
+        boolean customAapt = false;
+        String aaptPath = apkOptions.aaptPath;
+        List<String> cmd = new ArrayList<String>();
+
+        // path for aapt binary
+        if (! aaptPath.isEmpty()) {
```

Comment:
There is nearly 30 lines for just determining if we can use the aaptBinary. We should defer this logic to the aaptManager. It takes a file/string and determines if we can execute it, if so returns it. If null is return, we default to the "aapt"/"aapt2" command